apollo-compiler/build.gradle.kts

```diff
@@ -36,6 +32,12 @@ dependencies {
   testImplementation(golatac.lib("truth"))
   testImplementation(golatac.lib("kotlin.test.junit"))
   testImplementation(golatac.lib("google.testparameterinjector"))
+  testImplementation(project(":apollo-api-java")) {
+    because("For Java and Guava Optional adapter use in the Java generated code")
+  }
```

Comment:
I missed this in the last PR but this sounds a bit fishy. Because we relocate everything in the plugin, we must avoid reflection so I don't think we need this? Instead use `JavaClassNames.foo` like for other symbols?

Reply:
How about `Java generated code references Java and Guava Optionals`?

Reply:
Sounds good 👍